examples/patch-testing/program/src/main.rs

```diff
@@ -1,28 +1,54 @@
 #![no_main]
 sp1_zkvm::entrypoint!(main);
 
+use alloy_primitives::{address, hex, Signature};
 use curve25519_dalek_ng::edwards::CompressedEdwardsY;
-use ed25519_consensus::{Signature, VerificationKey};
+use ed25519_consensus::{Signature as Ed25519Signature, VerificationKey as Ed25519VerificationKey};
 use sha2_v0_10_6::{Digest as Digest_10_6, Sha256 as Sha256_10_6};
 // use sha2_v0_10_8::{Digest as Digest_10_8, Sha256 as Sha256_10_8};
 use sha2_v0_9_8::{Digest as Digest_9_8, Sha256 as Sha256_9_8};
+use std::str::FromStr;
 use tiny_keccak::{Hasher, Keccak};
 
-/// To add testing for a new patch, add a new case to the function below.
-fn main() {
-    let input = [1u8; 32];
-
-    let sig: Signature = sp1_zkvm::io::read();
-    let vk: VerificationKey = sp1_zkvm::io::read();
-    let msg: Vec<u8> = sp1_zkvm::io::read_vec();
-
-    // Test Keccak.
-    let mut hasher = Keccak::v256();
-    hasher.update(&input);
+/// Simple interface to the [`keccak256`] hash function.
+///
+/// [`keccak256`]: https://en.wikipedia.org/wiki/SHA-3
+fn keccak256<T: AsRef<[u8]>>(bytes: T) -> [u8; 32] {
     let mut output = [0u8; 32];
+    let mut hasher = Keccak::v256();
+    hasher.update(bytes.as_ref());
     hasher.finalize(&mut output);
+    output
+}
+
+/// Emits ED_ADD and ED_DECOMPRESS syscalls.
+fn test_ed25519_consensus() {
+    // Example signature and message.
+    let vk = hex!("9194c3ead03f5848111db696fe1196fbbeffc69342d51c7cf5e91c502de91eb4");
+    let msg = hex!("656432353531392d636f6e73656e7375732074657374206d657373616765");
+    let sig = hex!("69261ea5df799b20fc6eeb49aa79f572c8f1e2ba88b37dff184cc55d4e3653d876419bffcc47e5343cdd5fd78121bb32f1c377a5ed505106ad37f19980218f0d");
+
+    let vk: Ed25519VerificationKey = vk.try_into().unwrap();
+    let sig: Ed25519Signature = sig.into();
+    vk.verify(&sig, &msg).unwrap();
+}
+
+/// Emits ED_DECOMPRESS syscalls.
+fn test_curve25519_dalek_ng() {
+    let input = [1u8; 32];
+    let y = CompressedEdwardsY(input);
+    let _ = y.decompress();
+}
 
-    // Test SHA256.
+/// Emits KECCAK_PERMUTE syscalls.
+fn test_keccak() {
+    let input = [1u8; 32];
+    let _ = keccak256(input);
```

Comment:
would be good to validate this and the dalek_ng above return correct results

Reply:
Validated.